tracer/test/Datadog.Trace.Tests/SpanContextTests.cs

```diff
@@ -38,5 +40,109 @@ public void OverrideTraceIdWithParent()
             spanContext.SpanId.Should().Be(childSpanId);
             spanContext.TraceId.Should().Be(parentTraceId, "trace id shouldn't be overriden if a parent trace exists. Doing so would break the HttpWebRequest.GetRequestStream/GetResponse integration.");
         }
+
+        [Fact]
+        public void EnumerateKeys()
+        {
+            var context = CreateSpanContext();
+
+            // fail test if new keys are added
+            context.Keys.Should().HaveCount(12);
+
+            // fail tests if any key is missing
+            context.Keys.Should()
+                   .Contain(
+                        new[]
+                        {
+                            "__DistributedKey-TraceId",
+                            "__DistributedKey-ParentId",
+                            "__DistributedKey-SamplingPriority",
+                            "__DistributedKey-Origin",
+                            "__DistributedKey-RawTraceId",
+                            "__DistributedKey-RawSpanId",
+                            "__DistributedKey-PropagatedTags",
+                            "__DistributedKey-AdditionalW3CTraceState",
+                            "x-datadog-trace-id",
+                            "x-datadog-parent-id",
+                            "x-datadog-sampling-priority",
+                            "x-datadog-origin",
+                        });
+        }
```

Comment:
nit: Currently, if a new key is added, this will fail on the count, without telling you what's new. It's probably better to use something like `BeEquivalentTo` so it tells you what's changed? Same goes for the other new tests. (I get confused about the collection comparators, worth checking that works as expected! There's also `IntersectsWith`, `IsSubsetOf`, `ContainsKeys` etc, take your pick 😅)

Reply:
I couldn't find a method that asserted that the dictionary contain all keys, but no additional keys, while ignoring order.

Reply:
You could store this new dictionary in a variable and then do `IsSubsetOf` both ways:

```csharp
var expectedSet = new HashSet<string>() { }; // TODO: Add keys
context.Keys.IsSubsetOf(expectedSet);
expectedSet.IsSubsetOf(context.Keys);
```

Reply:
Nice, thank you! I also added a third check for the keys and pairs (but not values) list to ensure there are no duplicates.